### PR TITLE
feat: Evaluation DAO Kick Off 

### PR DESCRIPTION
**Pull request by nir1218**
# Description

The PR is to initiate a discussion regarding the evaluation DAO, #407, following a recent office hours meeting with @moul, see https://github.com/gnolang/meetings/blob/main/gor-office-hours/2023-04-25--gor-office-hours--evaluation-dao.jpg
By all means, the code is just to get the ball rolling with code rather than having a plain text contribution.

@moul To begin with, should I pick another location for the files?

# How has this been tested?

No testing was performed, code is just to start the effort.

### CODE REVIEW

**Comment by moul** [comment]
Thanks for the kickstart.

It's too early to decide on the location, but we can revisit that later. We can either integrate evaluation as a subpart of the Gnoland DAO or make it a top-level feature that other DAOs can use. We could also make it a reusable package in the future, but let's keep it simple for now and focus on one problem at a time.

Here's a suggestion: start writing tests soon as they will demonstrate your intentions and make it easier for others to review and provide feedback. Additionally, consider creating a genesis list of users for the evaluation DAO. They can onboard the first wave of users and conduct the initial reviews.

**Comment by moul** [comment]
I would like to propose a strategy for addressing complex topics where multiple proposals could legitimately be considered valid, especially during the exploration phase.
In such cases, it may be too much responsibility for everyone, including reviewers, to select the best approach early on.

My suggestion is to encourage the submission of multiple competing solutions to the repository, which can be reviewed side by side, forked, and extended.

I believe that each unique and approved (unique enough) proposal should receive its part of contribution points at the end.

While I welcome suggestions for improvement, I do not think this approach should be a general rule, but rather a ready-to-use pattern that can be applied as needed.

---

I suggest `gno.land/r/x/nir1218-evaluation-proposal` or `gno.land/r/x/nir1218/evaluation-v1` instead of `gno.land/r/x/evaluation/nir1218-proposal` to avoid limiting those who may want to implement both evaluation and governance at the same time or divide the topic.

File: examples/gno.land/r/x/nir1218-evaluation-proposal/contribution.gno
Comment (tbruyelle) [reply examples/gno.land/r/x/nir1218-evaluation-proposal/contribution.gno:5]: Actually, I don't understand the usage of `contributionStatus`, it really looks like a set of constants, so why using a map?

**Comment by nir1218** [comment]
> Just to note, we only accept underscores ("_") in package names. Please use 'git mv' to rename your package accordingly.
> 
> Any specific directions for reviewers?

Done
Not yet, still WIP

File: examples/gno.land/r/x/nir1218-evaluation-proposal/contribution.gno
Comment (nir1218) [reply examples/gno.land/r/x/nir1218-evaluation-proposal/contribution.gno]: > We currently have a bug when `map` is used in the store (see #311), the recommandation is to use `avl.Tree` instead.

Sure, I will try

Comment (nir1218) [reply examples/gno.land/r/x/nir1218-evaluation-proposal/contribution.gno]: > Actually, I don't understand the usage of `contributionStatus`, it really looks like a set of constants, so why using a map?

Using a map for a faster lookup rather than iterating match if the status is valid, if I use the avl.Tree then it would be fine too.

Comment (tbruyelle) [reply examples/gno.land/r/x/nir1218-evaluation-proposal/contribution.gno]: > Using a map for a faster lookup rather than iterating match if the status is valid, if I use the avl.Tree then it would be fine too.

Sorry, maybe I missed something, but if your keys are equals to your values, why using a map ? Also why storing it in the store ?

Comment (nir1218) [reply examples/gno.land/r/x/nir1218-evaluation-proposal/contribution.gno]: > > Using a map for a faster lookup rather than iterating match if the status is valid, if I use the avl.Tree then it would be fine too.
> 
> Sorry, maybe I missed something, but if your keys are equals to your values, why using a map ? Also why storing it in the store ?

The values in the map are for reference only to validate that the status update is from a given set of statuses. I could use a map with no values as I did in pull_request.gno. Sorry, maybe I don't explain my self well.

**Comment by moul** [comment]
As this is an initial example in a possible series, we'll stick to simple feedback for now. Perfection isn't required at this stage. Post-merge, @nir1218 can still submit revisions via PRs.

@nir1218, please resolve the CI issues. Thank you.

**Comment by nir1218** [comment]
> As this is an initial example in a possible series, we'll stick to simple feedback for now. Perfection isn't required at this stage. Post-merge, @nir1218 can still submit revisions via PRs.
> 
> @nir1218, please resolve the CI issues. Thank you.

@moul  CI issues resolved

File: examples/gno.land/r/x/nir1218_evaluation_proposal/EVALUATION.md
Comment (ajnavarro) [inline examples/gno.land/r/x/nir1218_evaluation_proposal/EVALUATION.md:5]:
```suggestion
This document describes general ideas regarding contributions evaluation, the principles laid out are intendant to be part of the Evaluation DAO.
```

File: examples/gno.land/r/x/nir1218_evaluation_proposal/EVALUATION.md
Comment (ajnavarro) [inline examples/gno.land/r/x/nir1218_evaluation_proposal/EVALUATION.md:25]:
```suggestion
Contributors DAO will designate members of a committee, in the beginning, the evaluation committee members will be the core development team members or any other trusted entity.
```

File: examples/gno.land/r/x/nir1218_evaluation_proposal/EVALUATION.md
Comment (ajnavarro) [inline examples/gno.land/r/x/nir1218_evaluation_proposal/EVALUATION.md:1]: I suggest using a grammar check on this document to find any misspellings.

Comment (nir1218) [reply examples/gno.land/r/x/nir1218_evaluation_proposal/EVALUATION.md:1]: Thanks for reviewing, my bad.
I have corrected the spelling in the GNO files as well.

**Review by moul — APPROVED** [review]
Pre-approving, the code is well located.
We can merge once @nir1218 requests, provided there are no additional comments to address.

**Comment by nir1218** [comment]
> Pre-approving, the code is well located. We can merge once @nir1218 requests, provided there are no additional comments to address.

I am fine with the initial merge, I will submit more changes on new PRs.

File: examples/gno.land/r/x/nir1218_evaluation_proposal/vote.gno
Comment (thehowl) [inline examples/gno.land/r/x/nir1218_evaluation_proposal/vote.gno:9]:
```suggestion
	VoteYes = "YES"
	VoteNo  = "NO"
```

**Review by thehowl — APPROVED** [review]
pre-approving with just a small nit

File: examples/gno.land/r/x/nir1218_evaluation_proposal/vote.gno
Comment (nir1218) [reply examples/gno.land/r/x/nir1218_evaluation_proposal/vote.gno]: Thanks @thehowl, your comment has been addressed.

**Comment by nir1218** [comment]
@MichaelFrazzy I would appreciate if you could look into the above PR regarding my initial thoughts about the Evaluation DAO. It has been a while since I worked on it, but if you see value in it, or even if you don't, I would like to collaborate with you and contribute as much as I can.

**Comment by MichaelFrazzy** [comment]
> @MichaelFrazzy I would appreciate if you could look into the above PR regarding my initial thoughts about the Evaluation DAO. It has been a while since I worked on it, but if you see value in it, or even if you don't, I would like to collaborate with you and contribute as much as I can.

Hey!! I'll begin going through everything, I really appreciate the time you've spent here. Quite a lot has been decided and worked on with Evaluation DAO since this was first created. More in terms of its design and eventual responsibilities opposed to its actual code implementation, so this seems like it'll be very helpful! 

Let's maybe touch base soon and I can fully catch you up there, and in the meantime I can catch up on this code! I'll either write out the details asap or we can hop on a call. I have a system for scraping, logging, and analyzing PRs and their content including on a username-wide level (contributor profiles) nearly done that could likely tie in here as well. Originally for the Game of Realms revamp and eventually Evaluation DAO.

All that being said, Evaluation DAO is looking like it isn't going to launch until after mainnet but it'll certainly be relieving to have it as good to go before then as possible. The closest thing we have to that in the meantime is core team judges that'll be set for Game of Realms, and while this will be testing many of the same systems we can for all intents and purposes view it as separate from post-launch Evaluation DAO still. Thanks again!

**Comment by nir1218** [comment]
Thanks @MichaelFrazzy for the quick response, when ever you are ready to have a call please DM me over Discord or get in touch on the server. I am willing to contribute in other areas as well if you need help with other stuff.

**Comment by MichaelFrazzy** [comment]
Sounds perfect! Could you please remind me what your name on Discord is? I'm MichaelFrazzy on there as well

**Comment by nir1218** [comment]
@MichaelFrazzy My name is nir1218 on Discord as well.

**Comment by MichaelFrazzy** [comment]
Weird, it wasn't showing up when I searched it haha. I must have been searching the wrong area, I'll find it thanks!

**Pull request merged**
